### PR TITLE
Add helpdesk functionality to register

### PR DESCRIPTION
**Pull request by onur011**
Now we save the register confirmation in our database for helpdesk client to use.

### CODE REVIEW

File: src/models/BankAccount.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  static all() {
+    return Model.all(Client.collection)
+  }
+
+  static create(data) {
+    return Model.create(Client.collection, Client._fields, data)
+  }
+  
+}
```
Comment (sebika) [inline src/models/BankAccount.js]: You should add a newline here

**Review by sebika — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
 
+.vscode/
```
Comment (Jokeswar) [inline .gitignore]: Keep the same style `/.vscode`

File: src/contexts/AuthContext.js [hunk 3]
```diff
@@ -15,6 +15,7 @@ export function AuthProvider({ children }) {
 
   async function register(data) {
     return auth.createUserWithEmailAndPassword(data.email, data.password).then(async (_) => {
+      await auth.signOut()
```
Comment (Jokeswar) [inline src/contexts/AuthContext.js:18]: Add a new line after this one. A bit of breathing room

File: src/contexts/AuthContext.js [hunk 4]
```diff
@@ -24,20 +25,28 @@ export function AuthProvider({ children }) {
...
-          id: currentUser.id
+          id: userQuery.docs[0].id
         }
       })
     })
   }
 
-  function login(email, password) {
+  async function login(email, password) {
+
```
Comment (Jokeswar) [inline src/contexts/AuthContext.js]: This new line is useless

File: src/contexts/AuthContext.js [hunk 5]
```diff
@@ -24,20 +25,28 @@ export function AuthProvider({ children }) {
...
+    const currentUser = (await Client.all().where('mail', '==', email).get()).docs[0]
+
```
Comment (Jokeswar) [inline src/contexts/AuthContext.js]: And this new line too

File: src/models/BankAccount.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  static collection = 'BankAccount'
+
+  static all() {
+    return Model.all(Client.collection)
+  }
+
+  static create(data) {
+    return Model.create(Client.collection, Client._fields, data)
+  }
+
```
Comment (Jokeswar) [inline src/models/BankAccount.js]: Extra new line. Big no no

File: src/contexts/AuthContext.js [hunk 7]
```diff
@@ -24,20 +25,28 @@ export function AuthProvider({ children }) {
...
+    const accountRequest = (await ClientAccountCreation.all().where('client.id', '==', currentUser.id).get())
+
+    if (accountRequest.size !== 0)
+      throw new Error('Request not accepted!')
```
Comment (Jokeswar) [inline src/contexts/AuthContext.js]: In the context of log in this message is a bit vague. Let's spice it up a bit! `The activation of your account has not been approved yet!`

**Review by Jokeswar — CHANGES_REQUESTED** [review]
Just one more push :smile:

**Review by sebika — APPROVED** [review]
Nice. Good job!

File: src/contexts/AuthContext.js (hunk 7 above)
Comment (onur011) [reply src/contexts/AuthContext.js]: i will find you ...

File: src/contexts/AuthContext.js [hunk 8]
```diff
@@ -24,20 +26,26 @@ export function AuthProvider({ children }) {
...
     })
   }
 
-  function login(email, password) {
+  async function login(email, password) {
+    const currentUser = (await Client.all().where('mail', '==', email).get()).docs[0]
+    const accountRequest = (await ClientAccountCreation.all().where('client.id', '==', currentUser.id).get())
+
+    if (accountRequest.size !== 0)
+      throw new Error('The account creation request has not been accepted yet! !')
```
Comment (Jokeswar) [inline src/contexts/AuthContext.js]: One `!` would be enough

**Review by Jokeswar — CHANGES_REQUESTED** [review]
Great!

**Review by Jokeswar — APPROVED** [review]

**Pull request merged**